thewalrus/quantum.py

```diff
@@ -962,6 +964,30 @@ def is_classical_cov(cov, hbar=2, atol=1e-08):
     return False
 
 
+def find_classical_subsystem(cov, hbar=2, atol=1e-08):
+    """Find the largest integer `k` so that subsystem in modes `[0,1,...,k-1]` is a classical state.

```

Comment:
What about if modes `[2, 3, .., k]` is a classical subsystem (e.g., there is some offset?)

Reply:
In general one could consider finding a classical subsystem in a any subset of the modes. This is a a combinatorially large space and thus extremely difficult to search. This is why I just search for this particular order. If the user wants to search for [2,3,...,k] they can permute the modes...